Description: Add multi-arch paths to virtuoso's plugins search path.
 This patch adds multi arch paths to the list of possible paths in where
 virtuoso plugins can be found.
 It does not adds this path if the build happens on a non-m-a aware
 environment.
Author: Maximiliano Curia <maxy@debian.org>

Index: soprano/CMakeLists.txt
===================================================================
--- soprano.orig/CMakeLists.txt	2013-10-26 21:38:41.206651307 +0200
+++ soprano/CMakeLists.txt	2013-10-26 21:38:41.202651468 +0200
@@ -136,6 +136,10 @@
 endif(NOT SOPRANO_DISABLE_VIRTUOSO_BACKEND)
 set(HAVE_DBUS BUILD_DBUS_SUPPORT)
 
+if(CMAKE_LIBRARY_ARCHITECTURE)
+    add_definitions(-DLIBRARY_ARCHITECTURE="${CMAKE_LIBRARY_ARCHITECTURE}")
+endif(CMAKE_LIBRARY_ARCHITECTURE)
+
 ##################  setup install directories  ################################
 set (LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
 set (LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Library directory name")
Index: soprano/soprano/sopranodirs.cpp
===================================================================
--- soprano.orig/soprano/sopranodirs.cpp	2013-10-26 21:38:41.206651307 +0200
+++ soprano/soprano/sopranodirs.cpp	2013-10-26 21:39:33.700543565 +0200
@@ -132,6 +132,10 @@
 #else
     paths << QLatin1String( "/usr/lib"SOPRANO_LIB_SUFFIX );
     paths << QLatin1String( "/usr/local/lib"SOPRANO_LIB_SUFFIX );
+#ifdef LIBRARY_ARCHITECTURE
+    paths << QLatin1String( "/usr/lib/"LIBRARY_ARCHITECTURE );
+    paths << QLatin1String( "/usr/local/lib/"LIBRARY_ARCHITECTURE );
+#endif
     paths += Soprano::envDirList( "LD_LIBRARY_PATH" );
 #endif
     return paths;
